client/src/components/post/ContentOptions.js

```diff
@@ -0,0 +1,60 @@
+import React, { useContext, useState } from 'react'
+
+import PostContext from '../../context/post/PostContext'
+
+import menu_icon from '../../assets/images/menu_icon.png'
+
+const ContentOptions = (props) => {
+  const postContext = useContext(PostContext)
+  const { setDeleteModal } = postContext
+
+  const { id, contentType } = props
+
+  const toggleModal = (e) => {
+    e.preventDefault()
+    setDeleteModal(id, contentType)
+    setShowMenu(!showMenu)
+  }
+
+  const [showMenu, setShowMenu] = useState(false)
```

Comment:
I know that javascript will hoist declarations of variables/functions to make everything ok but can we put this line above the `toggleModal` function on line 13, in other languages you wouldn't be able to use a function before it is declared